torch/_refs/__init__.py

```diff
@@ -1708,6 +1708,13 @@ def sub(
     """
 
     a, b = _maybe_broadcast(a, b)
+    
+    if isinstance(a, TensorLike) and isinstance(b, TensorLike):
+        torch._check(
+            not utils.is_boolean_dtype(a.dtype)
+            and not utils.is_boolean_dtype(b.dtype),
```

Comment:
In eager, either tensor being a bool is not allowed. We should update the check to:
`not (utils.is_boolean_dtype(a.dtype) or utils.is_boolean_dtype(b.dtype))`